Add Update the revoke inactive tokens after to GraphQL cookbook (#2306)
buildkite/docs: pages/apis/graphql/graphql_cookbook.md

}
```

> ### Update the revoke inactive tokens after

Reviewer: Ending on "after" here doesn't feel complete. How about:

```suggestion
### Update when inactive access tokens are revoked
```

> ### Update when inactive access tokens are revoked

Author: ```suggestion
### Update API access tokens revocation policy
```

@mbelton-buildkite Hello, we are planning to [name this feature](https://github.com/buildkite/buildkite/pull/13085#discussion_r1270106986), feel free to leave suggestion 🙏🏻

Reviewer: I'll have a look at that link. I'm 50/50 on that suggestion. While accurate, I think it sounds more complex than it needs to. Ideally, I'd like these headings to match the words someone might use to describe the task they want to do. They shouldn't have to know the feature name. 🤔

Have a think about that and I'm happy for you to update it if you think it helps more.

> On the Enterprise plan, you can control when inactive API access tokens are revoked. By default, they are never (`NEVER`) revoked, but you can set your token expiry policy to either 30, 60, 90, 180, or 365 days.

Reviewer: Probably best to update this if the feature name is changing:

```suggestion
On the Enterprise plan, you can control when inactive API access tokens are revoked. By default, they are never (`NEVER`) revoked, but you can set your token revocation policy to either 30, 60, 90, 180, or 365 days.
```

> You can control how many days to revoke inactive API access tokens. By default that's `NEVER` revoke inactive API access tokens, but you can set to 30, 60, 90, 180, 365 days. This is only available to Enterprise customers.

Reviewer: General tightening + I think it's good to be upfront about the Enterprise restriction:

```suggestion
On the Enterprise plan, you can control when inactive API access tokens are revoked. By default, they are never (`NEVER`) revoked, but you can set your token expiry policy to either 30, 60, 90, 180, or 365 days.
```

```graphql

> mutation UpdateRevokeInactiveTokenAfter {

Reviewer: I think it's better if the label can stand on its own. Maybe:

```suggestion
mutation UpdateRevokeInactiveTokenPeriod {
```

>   organizationRevokeInactiveTokensAfterUpdate(input: {

Reviewer: Is it too late to update the name of this? I'd go with `organizationRevokeInactiveTokenPeriodUpdate` orrr if it's not an issue to remove "organization" then `revokeInactiveTokenPeriodUpdate`.

Author: It is following the convention ([internal link](https://github.com/buildkite/buildkite/tree/main/app/models/api/graph/mutations)), so we may need to revisit the mutation naming if we really want to rename them!

Reviewer: Hmmmm I might be missing the convention you mean, but it looks like the current name doesn't match the others. The main convention I see is:

> [thing] + [action]

- clusterAgentToken + Update
- jobTypeCommand + Cancel
- pipelineSchedule + Delete

I see there are some exceptions, but most look to follow that pattern.

"organizationRevokeInactiveTokensAfter" is not a thing. Changing that part to "organizationRevokeInactiveTokensPeriod" would make it a thing and follow the convention.

Does that make sense?

Author: > [thing] + [action]

Yup.

`OrganizationRevokeInactiveTokensAfter` + `UpdateMutation` same as
`OrganizationAPIIPAllowlist` + `UpdateMutation`. 

They all started with `Organization[thing]` + `action`.

The field is not called `Period` [on UI](https://buildkite.com/organizations/~/security), so was keeping the same as the UI. If you strongly feel we should rename all to Period (which is possible), could you raise an escalation on Foundation team? Thanks!

Reviewer: Got it, thanks! I'll raise an escalation to see how others feel. 😊
